pkg/sources/indices.go

```diff
@@ -141,7 +141,12 @@ func (fi *feedIndex) directoryLocations(r io.Reader) ([]location, error) {
 			return nil, fmt.Errorf("column 2 in line %d is not a valid RFC3339 time: %w", lineNo, err)
 		}
 		if !doc.IsAbs() {
-			doc = fi.base.ResolveReference(doc)
+			// Use copy to avoid side effects
+			base := *fi.base
+			if !strings.HasSuffix(base.Path, "/") {
```

Comment:
Should be better done in source.go before line 224 to move that out of the loop. That also would help the rolie path, which may be affected by this, too.